realm/src/main/java/io/realm/internal/Context.java

```diff
@@ -55,6 +60,12 @@ public void executeDelayedDisposal() {
                 TableQuery.nativeClose(nativePointer);
             }
             abandonedQueries.clear();
+
+            NativeObjectReference reference;
+            while ((reference = (NativeObjectReference) rowReferenceQueue.poll()) != null) {
```

Comment:
Not possible but not for the reason you mention.
Native destructors **can** be called from other threads, but the constraint is more subtle.
You cannot create and delete native objects belonging to the same context (shared group) concurrently. This is what prevents the creation of a worker thread.